wren-ai-service/src/pipelines/generation/sql_answer.py

```diff
@@ -49,166 +40,109 @@
-
-
 @timer
 @observe(capture_input=False)
 def prompt(
     query: str,
     sql: str,
-    execute_sql: dict,
+    sql_data: dict,
     language: str,
     prompt_builder: PromptBuilder,
 ) -> dict:
     return prompt_builder.run(
         query=query,
         sql=sql,
-        sql_data=execute_sql["results"],
+        sql_data=sql_data,
         language=language,
     )
 
 
 @async_timer
 @observe(as_type="generation", capture_input=False)
-async def generate_answer(prompt: dict, generator: Any) -> dict:
-    return await generator.run(prompt=prompt.get("prompt"))
-
-
-@timer
-@observe(capture_input=False)
-def post_process(
-    generate_answer: dict, post_processor: SQLAnswerGenerationPostProcessor
-) -> dict:
-    return post_processor.run(generate_answer.get("replies"))
+async def generate_answer(prompt: dict, generator: Any, query_id: str) -> dict:
+    return await generator.run(prompt=prompt.get("prompt"), query_id=query_id)
 
 
 ## End of Pipeline
 
 
-class AnswerResults(BaseModel):
-    reasoning: str
-    answer: str
-
-
-SQL_ANSWER_MODEL_KWARGS = {
-    "response_format": {
-        "type": "json_schema",
-        "json_schema": {
-            "name": "sql_summary",
-            "schema": AnswerResults.model_json_schema(),
-        },
-    }
-}
+SQL_ANSWER_MODEL_KWARGS = {"response_format": {"type": "text"}}
 
 
 class SQLAnswer(BasicPipeline):
     def __init__(
         self,
         llm_provider: LLMProvider,
-        engine: Engine,
         **kwargs,
     ):
+        self._user_queues = {}
         self._components = {
-            "data_fetcher": DataFetcher(engine=engine),
             "prompt_builder": PromptBuilder(
                 template=sql_to_answer_user_prompt_template
             ),
             "generator": llm_provider.get_generator(
                 system_prompt=sql_to_answer_system_prompt,
                 generation_kwargs=SQL_ANSWER_MODEL_KWARGS,
+                streaming_callback=self._streaming_callback,
             ),
-            "post_processor": SQLAnswerGenerationPostProcessor(),
         }
 
         super().__init__(
             AsyncDriver({}, sys.modules[__name__], result_builder=base.DictResult())
         )
 
+    def _streaming_callback(self, chunk, query_id):
+        if query_id not in self._user_queues:
+            self._user_queues[
+                query_id
+            ] = asyncio.Queue()  # Create a new queue for the user if it doesn't exist
+        # Put the chunk content into the user's queue
+        asyncio.create_task(self._user_queues[query_id].put(chunk.content))
+        if chunk.meta.get("finish_reason") == "stop":
+            asyncio.create_task(self._user_queues[query_id].put("<DONE>"))
+
+    async def get_streaming_results(self, query_id):
+        if query_id not in self._user_queues:
+            self._user_queues[
+                query_id
+            ] = asyncio.Queue()  # Ensure the user's queue exists
```

Comment:
im curious about create an queue for it here, because i think raise an exception will be better if not found some resource